my-app/src/app/courses-page/courses-page.component.ts

```diff
@@ -16,40 +14,35 @@ import { CoursesService } from './services';
 export class CoursesPageComponent implements OnInit, OnDestroy {
   public foundCourses: Course[];
   public searchPhrase: string;
-  public courses$: Observable<Course[]>;
-  public userData$: Observable<UserInfo>;
+  public courses$: Observable<Course[]> = this.contextStoreFacadeService.selectCoursesList();
+  public userData$: Observable<UserInfo> = this.contextStoreFacadeService.selectUserData();
+  public isOnline$: Observable<boolean> = this.contextStoreFacadeService.selectIsOnline();
   private unsubscribe: Subject<void> = new Subject();
 
-  constructor(private coursesService: CoursesService, private authService: AuthenticationService) {}
+  constructor( private contextStoreFacadeService: ContextStoreFacadeService) {}
 
 
   ngOnInit(): void {
-    this.courses$ = this.coursesService.getCoursesList();
-
-    this.userData$ = this.authService.getUserData();
+    
   }
 
   onDeleteCourse(course: Course): void {
-    this.coursesService.removeCourseItem(course).pipe(
-      takeUntil(this.unsubscribe)
-    ).subscribe(
-      () => {
-        this.courses$ = this.searchPhrase 
-          ? this.coursesService.getSearchedCourses(this.searchPhrase) 
-          : this.coursesService.getCoursesList();
-      },
-      (err) => console.error(err),
-    )
+    if (Boolean(course.id)) {
+      this.contextStoreFacadeService.dispatchDeleteCourse({ ...course });
+    }
+    this.contextStoreFacadeService.dispatchFetchCoursesList();
   }
 
   onLoadMoreCourses(): void {
-    console.log('Load more courses');
-    this.courses$ = this.coursesService.loadMoreCourses();
+    this.contextStoreFacadeService.dispatchLoadMoreCourses();
   }
 
   onCourseSearch(searchPhrase: string): void {
-    this.searchPhrase = searchPhrase;
-    this.courses$ = this.coursesService.getSearchedCourses(searchPhrase);
+    this.contextStoreFacadeService.dispatchSearchCourses(searchPhrase);
```

Comment:
you don't need to dispatch dispatchSearchCourses at all if searchPhrase === '' is true

Reply:
fixed